src/NHibernate/Logging.cs

```diff
@@ -398,16 +544,28 @@ static Log4NetLoggerFactory()
 		[Obsolete("Use this as an INHibernateLoggerFactory instead.")]
 		public IInternalLogger LoggerFor(string keyName)
 		{
-			return new Log4NetLogger(GetLoggerByNameDelegate(typeof(Log4NetLoggerFactory).Assembly, keyName));
+			INHibernateLoggerFactory nhFact = this;
+			return nhFact.LoggerFor(keyName);
 		}
 
 		[Obsolete("Use this as an INHibernateLoggerFactory instead.")]
 		public IInternalLogger LoggerFor(System.Type type)
 		{
-			return new Log4NetLogger(GetLoggerByTypeDelegate(type));
+			INHibernateLoggerFactory nhFact = this;
+			return nhFact.LoggerFor(type);
 		}
 #pragma warning restore 618
 
+		INHibernateLogger INHibernateLoggerFactory.LoggerFor(string keyName)
```

Comment:
Once the old interface dropped, we could turn it to implicit implementation. (Same for next.)